tools/releases/version.sh

```diff
@@ -1,25 +1,29 @@
 #!/usr/bin/env bash
 
 # Kuma version is built as follows:
-# 1) If a git tag is present on the current commit, then the version is a git tag
-# 2) If the branch starts with "release" (like "release-1.3"), then the version is "$lastGitTag-$shortHash" (for example: 1.4.1-450174242)
-#    At the same time, if $lastGitTag is not final tag (for example 1.4.0-rc1 or 1.4.0-preview1), the version is "dev-$shortHash"
-# 3) If the branch does not start with "release", then the version is "dev-$shortHash" (for example: dev-450174242)
-
-lastGitTag=$(git describe --abbrev=0 --tags 2> /dev/null)
-shortHash=$(git rev-parse --short HEAD 2> /dev/null)
-currentBranch=$(git rev-parse --abbrev-ref HEAD 2> /dev/null)
+# 1) If a git tag is present on the current commit, then the version is a git tag (without a starting v)
+# 2) If the branch is "release-X.Y" look at the existing tags and use either X.Y.0+<shortHash> if there's none or
+#     the latest tag with a patch version increased by one (.e.g if latest tag is X.Y.1 the version will be `X.Y.2+<shortHash>`)
+# 3) In non release branch use `0.0.0-dev+$shortHash`
```

Comment:
Why not use the last git tag here?

Reply:
Because the last git tag is an already released version which is not even on this branch.

Reply:
But that can be another discussion

Reply:
Yes I like the fact that it's obvious which branch this version is coming from 0.0.0 makes it clear it's not from a release-X.Y branch